### PR TITLE
Closes #1083: Flaky test on engine-gecko-[beta|nightly].....

### LINKED ISSUE #1083: Flaky test on engine-gecko-nightly GeckoEngineSessionTest.kt:272
This test fun `saveState()` have been failing intermittently  GeckoEngineSessionTest.kt:272
```
:support-base:packageDebugAssets
:support-ktx:mergeDebugShaders
:support-ktx:compileDebugShaders
:support-ktx:generateDebugAssets
:support-ktx:packageDebugAssets
:support-utils:mergeDebugShaders
:support-utils:compileDebugShaders
:support-utils:generateDebugAssets
:support-utils:packageDebugAssets
:browser-engine-gecko-nightly:mergeDebugAssets
:browser-engine-gecko-nightly:generateDebugUnitTestConfig
:browser-engine-gecko-nightly:mockableAndroidJar
:browser-engine-gecko-nightly:processDebugUnitTestJavaRes
:browser-engine-gecko-nightly:transformClassesAndResourcesWithPrepareIntermediateJarsForDebug
:browser-engine-gecko-nightly:testDebugUnitTest

mozilla.components.browser.engine.gecko.GeckoEngineSessionTest > saveState FAILED
    java.lang.UnsatisfiedLinkError at GeckoEngineSessionTest.kt:272
42 tests completed, 1 failed
:browser-engine-gecko-nightly:testDebugUnitTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':browser-engine-gecko-nightly:testDebugUnitTest'.
> There were failing tests. See the report at: file:///build/android-components/components/browser/engine-gecko-nightly/build/reports/tests/testDebugUnitTest/index.html
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 5m 47s
265 actionable tasks: 261 executed, 4 up-to-date
[taskcluster 2018-10-17 14:29:09.572Z] === Task Finished ===
[taskcluster 2018-10-17 14:29:09.572Z] Unsuccessful task run with exit code: 1 completed in 559.804 seconds
```

### PR DESCRIPTION
**Pull request by csadilek**
Worth trying / verifying. Long-term we need a different solution anyway.

Trying a crazy thing to prevent `disposeNative` being called which causes the `UnsatisfiedLinkError`. See #1083 for details.

### CODE REVIEW

**Comment by csadilek** [comment]
The latest changes appear to workaround the issue. Ran this a couple of times now and can't reproduce the problem anymore, whereas it fails pretty reliably for other PRs now that both beta and nightly are affected.

**Comment by csadilek** [comment]
A cleaner way of doing this would be to expose our saveState runnable. Then we could check if msg.getCallback is our runnable, instead of ignoring everything but the first message.

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt [hunk 1]
```diff
@@ -262,13 +269,22 @@ class GeckoEngineSessionTest {
...
+        var first = true
+        ThreadUtils.sGeckoHandler = object : Handler() {
+            override fun sendMessageAtTime(msg: Message?, uptimeMillis: Long): Boolean {
+                // Only run the saveState runnable and ignore everything that comes after
+                if (first) {
+                    first = false
+                    return super.sendMessageAtTime(msg, uptimeMillis)
+                }
+                return true
+            }
```
Comment (pocmo) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt]: Is this handler implementation something we could stuff into `support-test`?

Comment (pocmo) [reply components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt]: I'll merge nevertheless to unblock other PRs. :)

**Pull request merged**